### PR TITLE
Project skeleton

### PR DESCRIPTION
**Pull request by danielcooperxyz**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.17-alpine AS build
+
+WORKDIR /wasp-simple-h264-payload-parser
+COPY . .
+RUN CGO_ENABLED=0 go build -o /bin/wasp-simple-h264-payload-parser
+
+FROM alpine
+COPY --from=build /bin/wasp-simple-h264-payload-parser /bin/wasp-simple-h264-payload-parser
```
Comment (dblane-digicatapult) [inline Dockerfile:8]: `RUN apk --no-cache add ffmpeg`

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]
Add README.md
Add LICENSE

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - KAFKA_CFG_ADVERTISED_LISTENERS=INTERNAL://kafka:9092,EXTERNAL://localhost:9092
+      - KAFKA_INTER_BROKER_LISTENER_NAME=INTERNAL
+    depends_on:
+      - zookeeper
+  kafka-topics:
+    image: "docker.io/bitnami/kafka:2-debian-10"
+    command: /bin/sh -c "
+      sleep 20 &&
+      kafka-topics.sh --bootstrap-server kafka:9092 --create --topic payloads.simpleH264 --partitions 1 --replication-factor 1 &&
+      kafka-topics.sh --bootstrap-server kafka:9092 --create --topic events --partitions 1 --replication-factor 1"
```
Comment (dblane-digicatapult) [inline docker-compose.yml]: `--topic video`

File: integration/message_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	"go.uber.org/zap"
+
+	"github.com/digicatapult/wasp-simple-h264-payload-parser/kafka"
+	"github.com/digicatapult/wasp-simple-h264-payload-parser/wasp"
+)
+
+func TestWaspTransformRelay(t *testing.T) {
+	inTopicName := "videos"
+	outTopicName := "videos-transformed"
+	kafkaBrokers := []string{"localhost:9092"}
```
Comment (dblane-digicatapult) [inline integration/message_test.go]: Are you sure?

Surely intopic name is `payloads.simpleH264` and outTopic should be `video` ?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+
+	"github.com/digicatapult/wasp-simple-h264-payload-parser/kafka"
+	"github.com/digicatapult/wasp-simple-h264-payload-parser/wasp"
+)
+
+func main() {
+	inTopicName := "videos"
+	outTopicName := "videos-transformed"
```
Comment (dblane-digicatapult) [inline main.go]: Same as above

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]
Could also do with making the InTopicName configurable.

OutTopicName will be static at `video` as this will be a common queue.

**Review by darrylmorton-digicatapult — APPROVED** [review]

File: main.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+const (
+	// InTopicNameKey defines the environment variable key for InTopicName
+	InTopicNameKey = "IN_TOPIC_NAME_KEY"
+
+	// OutTopicNameKey defines the environment variable key for OutTopicName
+	OutTopicNameKey = "OUT_TOPIC_NAME_KEY"
+
+	// KafkaBrokersKey defines the environment variable key for KafkaBrokers
+	KafkaBrokersKey = "KAFKA_BROKERS"
+)
```
Comment (dblane-digicatapult) [inline main.go]: Why are these not in `utils/env.go` ?

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: main.go (hunk 5 above)
Comment (danielcooperxyz) [reply main.go]: will move them

File: main.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+	inTopicName := util.GetEnv(util.InTopicNameKey, "payloads.simpleH264")
+	outTopicName := util.GetEnv(util.OutTopicNameKey, "video")
+	kafkaBrokersVar := util.GetEnv(util.KafkaBrokersKey, "localhost:9092")
+	kafkaBrokers := strings.Split(kafkaBrokersVar, ",")
+
+	cfg := zap.NewDevelopmentConfig()
+	if os.Getenv("ENV") == "production" {
+		cfg = zap.NewProductionConfig()
+
+		lvl, err := zap.ParseAtomicLevel(os.Getenv("LOG_LEVEL"))
```
Comment (dblane-digicatapult) [inline main.go]: `util`

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

**Review by dblane-digicatapult — APPROVED** [review]

**Pull request merged**